cmd/registry/cmd/compute/conformance.go

```diff
@@ -0,0 +1,208 @@
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/names"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/proto"
+)
+
+func conformantsCommand(ctx context.Context) *cobra.Command {
+	var linter string
+	cmd := &cobra.Command{
+		Use:   "conformance",
+		Short: "Compute lint results for API specs",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			filter, err := cmd.Flags().GetString("filter")
+			if err != nil {
+				log.Fatalf("Failed to get filter from flags: %s", err)
+			}
+
+			client, err := connection.NewClient(ctx)
+			if err != nil {
+				log.Fatalf("%s", err.Error())
+			}
+
+			name := args[0]
+
+			// Ensure that the provided argument is a spec
+			var specSegments []string
+			if specSegments = names.SpecRegexp().FindStringSubmatch(name); specSegments == nil {
+				log.Fatalf(
+					fmt.Sprintf(
+						"The provided argument %s does not match the regex of a spec",
+						name,
+					),
+				)
+			}
+			spec, err := names.ParseSpec(name)
+			if err != nil {
+				log.Fatalf("%s", err.Error())
+			}
+
+			projectSegments := []string{"projects", spec.ProjectID}
+
+			linterNameToLinter := make(map[string]conformance.Linter)
+			err = core.ListArtifacts(ctx, client, projectSegments, filter, true, func(artifact *rpc.Artifact) {
+				// Only consider artifacts which have the styleguide mimetype
+				messageType, err := core.MessageTypeForMimeType(artifact.GetMimeType())
+				if err != nil {
+					return
+				}
+				if messageType != "google.cloud.apigee.registry.applications.v1alpha1.styleguide" {
+					return
+				}
+
+				// Unmarshal the contents of the artifact into a style guide
+				styleGuide := &rpc.StyleGuide{}
+				err = proto.Unmarshal(artifact.GetContents(), styleGuide)
+				if err != nil {
+					return
+				}
+
+				// Construct a mapping between the linter name and the linter, and populate
+				// all the rules that the linter should support
+				for _, guideline := range styleGuide.GetGuidelines() {
+					for _, rule := range guideline.GetRules() {
+						linter_name := rule.GetLinter()
+						if _, ok := linterNameToLinter[linter_name]; !ok {
+							linter, err := conformance.CreateLinter(rule.GetLinter())
+							if err != nil {
+								// If the linter is unsupported, there is no reason
+								// to prematurely exit. We can just ignore this specific
+								// linter and log the message to the user.
+								log.Printf("%s", err.Error())
+							}
+							linterNameToLinter[linter_name] = linter
+						}
+
+						// Add each rule to the linter
+						linter := linterNameToLinter[linter_name]
+						for _, allowedMimeType := range styleGuide.MimeTypes {
+							linter.AddRule(allowedMimeType, rule.GetLinterRulename())
+						}
+					}
+
+				}
+			})
+
+			if err != nil {
+				log.Fatalf("%s", err.Error())
+			}
+
+			// Initialize task queue.
+			taskQueue, wait := core.WorkerPool(ctx, 16)
```

Comment:
If a task returns an error, how does the error get handled?

Reply:
The worker that executes this task selects on either a successfully `Done` context or an error. In the case of an error, the task logs the message to standard error, but does not drop/penalize other tasks that are already running.

For info:
https://github.com/apigee/registry/blob/05a9904c6d039605c5bd993968f2796a9d079e46/cmd/registry/core/tasks.go#L59-L63